src/main/java/net/mcreator/ui/dialogs/imageeditor/ResizeDialog.java

```diff
@@ -43,29 +45,40 @@ public ResizeDialog(MCreator window, Canvas canvas, Layer layer, VersionManager
 		JSpinner width = new JSpinner(new SpinnerNumberModel(layer.getWidth(), 0, 10000, 1));
 		JSpinner height = new JSpinner(new SpinnerNumberModel(layer.getHeight(), 0, 10000, 1));
 		JCheckBox type = new JCheckBox();
+		JCheckBox affectCanvas = new JCheckBox();
 
-		JButton cancel = new JButton(UIManager.getString("OptionPane.cancelButtonText"));
 		JButton ok = L10N.button("action.common.resize");
+		JButton cancel = new JButton(UIManager.getString("OptionPane.cancelButtonText"));
 		ok.setBackground((Color) UIManager.get("MCreatorLAF.MAIN_TINT"));
 		ok.setForeground((Color) UIManager.get("MCreatorLAF.BLACK_ACCENT"));
 		getRootPane().setDefaultButton(ok);
 
 		GridBagConstraints layoutConstraints = new GridBagConstraints();
 
-		cancel.addActionListener(e -> setVisible(false));
-
 		ok.addActionListener(e -> {
 			layer.resize((int) width.getValue(), (int) height.getValue(), type.isSelected());
-			versionManager.addRevision(new Modification(canvas, layer));
+			if (affectCanvas.isSelected() && (canvas.getWidth() < layer.getWidth()
+					|| canvas.getHeight() < layer.getHeight())) {
+				UUID uuid = UUID.randomUUID();
+				versionManager.addRevision(new Modification(canvas, layer).setUUID(uuid));
+				canvas.setSize(Math.max(canvas.getWidth(), layer.getWidth()),
+						Math.max(canvas.getHeight(), layer.getHeight()), uuid);
+			} else {
+				versionManager.addRevision(new Modification(canvas, layer));
+			}
 			setVisible(false);
 		});
 
+		cancel.addActionListener(e -> setVisible(false));

```

Comment:
Moving this doesn't change the behaviour. Please move it back.

Reply:
I just thought that <kbd>Cancel</kbd> button definition should normally follow <kbd>Ok</kbd> one's, not vice versa.

Reply:
I'll let it be this time, but in the future please make such changes a separate PR as this falls under refactoring and should be avoided to allow multiple people to work on the same files without complicated merge parties.